.github/workflows/check.yml

```diff
@@ -63,8 +63,15 @@ jobs:
                   components: rust-src, rustfmt, clippy
                   default: true
 
+            - name: Cache evcxr
+              id: cache-evcxr
+              uses: actions/cache@v2
+              with:
+                  path: ~/.cargo/bin/evcxr
+                  key: ${{ runner.os }}-cache-evcxr-${{ matrix.rust-version }}
```

Comment:
Looks like it can prevent updating `evcxr` to new version. Currently, we don't care about it and uses latest version. But if we cache some version of `evcxr`, CI will use cache and potentially will never user a new version.
Probably, we should specify version of `evcxr` for cargo (via `--version` flag according to [documentation](https://doc.rust-lang.org/cargo/commands/cargo-install.html)) and use it in key

Reply:
The cache depends on the `rustc` version, so it will be updated from time to time. But it's true that we could specify it explicitly, although I'm thinking that it will get relatively complex for a change that doesn't speed up the bottleneck (windows).

Reply:
I think the best approach would be:
* Restore evcxr from cache
* Always run `cargo --install evcxr` (if `evcxr` from cache has latest version, it will not be recompiled)
* Always save cache (that is override existing cache)

Unfortunatelly it is not possible now to [update cache](https://github.com/actions/cache/issues/342). We can try to use [a workaround](https://github.com/actions/cache/issues/342#issuecomment-673371329)

Reply:
I tried your suggestion, it seems to work.